Lecture-6/Task1/l6task1/Application.java

```diff
@@ -0,0 +1,139 @@
+    private static int[] packPrice = {10, 20, 30, 40, 50};
+    private static String[] brands = {"Tchibo", "Pele", "Nescafe", "Jacobs", "Maxwell"};
+    private static Van van;
+
+    public static String getRandomProductBrand() {
+        return brands[(int) (Math.random() * 5)];
+    }
+
+    public static List<Pack> createPacks() {
+        List<Pack> packs = new ArrayList<Pack>();
+        for (int i = 0; i < packCapacity.length; i++) {
+            Pack canPack = new Pack(packType[0], packValue[i], packPrice[i], packCapacity[i]);
+            packs.add(canPack);
+            Pack bagPack = new Pack(packType[1], packValue[i], 2 * packPrice[i], packCapacity[i]);
+            packs.add(bagPack);
+        }
+        return packs;
+    }
+
+    public static List<Coffee> createCoffee() {
+        List<Coffee> coffees = new ArrayList<Coffee>();
+        for (int i = 0; i < coffeeType.length; i++) {
+            Coffee coffeeArabica = new Coffee(coffeeType[i], coffeeSort[0], coffeePrice[i]);
+            Coffee coffeeRobusta = new Coffee(coffeeType[i], coffeeSort[1], 2 * coffeePrice[i]);
+            coffees.add(coffeeArabica);
+            coffees.add(coffeeRobusta);
+        }
+        return coffees;
+    }
+
+    public static List<Product> createProducts(List<Pack> packs, List<Coffee> coffees) {
+        List<Product> products = new ArrayList<>();
+        for (Pack pack : packs) {
+            for (Coffee coffee : coffees) {
+                Product product = new Product(coffee, pack, getRandomProductBrand());
+                products.add(product);
+            }
+        }
+        return products;
+    }
+
+    public static void showCoffees(List<Coffee> coffees) {
+        for (Coffee coffee : coffees) {
+            System.out.println(coffee.toString());
+        }
+    }
+
+    public static void showPacks(List<Pack> packs) {
+        for (Pack pack : packs) {
+            System.out.println(pack.toString());
+        }
+    }
+
+    public static void showProducts(List<Product> products) {
+        for (Product product : products) {
+            System.out.println(product.toString());
+        }
+    }
+
+    public static int getSummaryProductsValue(List<Product> products) {
+        int summaryProductsValue = 0;
+        for (Product product : products) {
+            summaryProductsValue += product.getValue();
+        }
+        return summaryProductsValue;
+    }
+
+    public static int getSummaryProductsPrice(List<Product> products) {
+        int summaryProductPrice = 0;
+        for (Product product : products) {
+            summaryProductPrice += product.getPrice();
+        }
+        return summaryProductPrice;
+    }
+
+    public static void fillVan(List<Product> products, int money) {
+        Collections.shuffle(products);
+        List<Product> currentProducts = new ArrayList<>();
+        int currentVanValue = 0;
+        for (Product product : products) {
+            money -= product.getPrice();
```

Comment:
Filling logic is not complete: another product may satisfy condition. E.g. you have 99 money left, so 100's product doesn't fit, but 50 is ok.
The same applies to van's capacity

Reply:
Logic of the method is changed.